### ISSUE TITLE
[a11y]: DatePicker [Range With Calendar] - year edit spinbox bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15252]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.38.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
Not discoverable with the JAWS in application mode

In application mode, the control is not identified by JAWS in Arrow key navigation.

### Recommended Fix
Make the control SR discoverable and SR navigable

When the screen reader doesn’t identify the page elements, it impacts the experience of users who are blind or have vision impairment. They can miss important information or fail to perform an action on the web page. 

### WCAG 2.1 Violation
1.3.1 Info and Relationships
